sklearn/tree/tests/test_monotonic_tree.py

```diff
@@ -0,0 +1,348 @@
+
+
+def test_multiple_output_raises():
+    X = [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]]
+    y = [[1, 0, 1, 0, 1], [1, 0, 1, 0, 1]]
+
+    for name, TreeClassifier in CLF_TREES.items():
+        est = TreeClassifier(
+            max_depth=None, monotonic_cst=np.array([-1, 1]), random_state=0
+        )
+        msg = "Monotonic constraints are not supported with multiple output"
+        with pytest.raises(ValueError, match=msg):
+            est.fit(X, y)
+
+
+def test_bad_monotonic_cst_raises():
+    X = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]]
+    y = [1, 0, 1, 0, 1]
+
+    for name, TreeClassifier in CLF_TREES.items():
+        msg = "monotonic_cst has shape 3 but the input data X has 2 features."
+        est = TreeClassifier(
+            max_depth=None, monotonic_cst=np.array([-1, 1, 0]), random_state=0
+        )
+        with pytest.raises(ValueError, match=msg):
+            est.fit(X, y)
+
+        msg = "monotonic_cst must be None or an array-like of -1, 0 or 1."
+        est = TreeClassifier(
+            max_depth=None, monotonic_cst=np.array([-2, 2]), random_state=0
+        )
+        with pytest.raises(ValueError, match=msg):
+            est.fit(X, y)
+
+        est = TreeClassifier(
+            max_depth=None, monotonic_cst=np.array([-1, 0.8]), random_state=0
+        )
+        with pytest.raises(ValueError, match=msg):
+            est.fit(X, y)
+
+
+def assert_1d_reg_tree_children_monotonic_bounded(tree_, monotonic_sign):
+    # Flip values to always check for increasing constraint
+    values = monotonic_sign * tree_.value
+    for i in range(tree_.node_count):
+        if tree_.children_left[i] > i and tree_.children_right[i] > i:
+            # Check monotonicity on children
+            i_left = tree_.children_left[i]
+            i_right = tree_.children_right[i]
+            assert values[i_left] <= values[i_right]
+            val_middle = values[i]
+            # Check bounds on grand-children, filtering out leaf nodes
+            if tree_.feature[i_left] >= 0:
+                i_left_right = tree_.children_right[i_left]
+                assert values[i_left_right] <= val_middle
+            if tree_.feature[i_right] >= 0:
+                i_right_left = tree_.children_left[i_right]
+                assert val_middle <= values[i_right_left]
+
+
+def assert_1d_reg_monotonic(clf, monotonic_sign, min_x, max_x, n_steps):
+    X_grid = np.linspace(min_x, max_x, n_steps).reshape(-1, 1)
+    y_pred_grid = clf.predict(X_grid)
+    assert (monotonic_sign * np.diff(y_pred_grid) >= 0.0).all()
+
+
+@pytest.mark.parametrize("monotonic_sign", (-1, 1))
+@pytest.mark.parametrize("splitter", ("best", "random"))
+@pytest.mark.parametrize("depth_first", (True, False))
+def test_1d_tree_nodes_values(
+    monotonic_sign, splitter, depth_first, global_random_seed
+):
+    # Adaptation from test_nodes_values in test_montonic_constraints.py
+    # Build a single tree with only one feature, and make sure the nodes
```

Comment:
```suggestion
    # Build a single tree with only one feature, and make sure the node
```

Reply:
The are several nodes in this tree.

Reply:
"make sure the node values" is already plural. Otherwise, it should be "nodes' values", but never "nodes values", in my limited understanding of english grammar.

Reply:
How about?
```suggestion
    # Build a single tree with only one feature, and make sure the nodes'
```

Reply:
I re-read some English grammar online on compound nouns and I was wrong and @lorentzenchr is correct. @jjerphan is probably also correct but let's god with the simple solution :)